scheduler/src/cook/mesos/util.clj

```diff
@@ -438,3 +439,32 @@
   [instance]
   (some #{(:instance/status instance)} #{:instance.status/running
                                          :instance.status/unknown}))
+
+(defn close-when-ch!
+  "When the value passed in is a channel, close it. Otherwise, do nothing"
+  [maybe-ch]
+  (try
+    (async/close! maybe-ch)
+    (catch Exception e)))
+
+(defn chime-at-ch
+  "Like chime-at (from chime[https://github.com/jarohen/chime])
+   but pass in an arbitrary chan instead of times to make a chime chan
+   Passes whatever comes off ch to f
+
+   Will try to close the item pulled from ch once f has completed if the item is a channel"
+  [ch f & [{:keys [error-handler on-finished]
+            :or {error-handler identity
+                 on-finished #()}}]]
+  (async/go-loop []
+                 (if-let [x (async/<! ch)]
+                   (do (async/<! (async/thread
+                                   (try
+                                     (f x)
+                                     (catch Exception e
+                                       (error-handler e)))))
+                       (close-when-ch! x)
```

Comment:
Currently no, though it is possible. This was a bit of a hack to try to support the old chime logic but get completion notification semantics. I will pass current time instead of x into f since that is closer to the original chime contract anyway.